File: src/main/java/com/example/eyeofsauron/service/SecuredFacilityService.java
package com.example.eyeofsauron.service;

import com.example.eyeofsauron.entity.SecuredFacility;
import com.example.eyeofsauron.repository.SecuredFacilityRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class SecuredFacilityService {
    private SecuredFacilityRepository securedFacilityRepository;

    public List<SecuredFacility> getAll() { return securedFacilityRepository.findAll(); }

    public Optional<SecuredFacility> getById(Long id) { return securedFacilityRepository.findById(id); }

    public void create(SecuredFacility securedFacility) { securedFacilityRepository.save(securedFacility); }

    public void update(SecuredFacility securedFacility) { securedFacilityRepository.save(securedFacility); }

    public void delete(SecuredFacility securedFacility) { securedFacilityRepository.delete(securedFacility); }

    public void deleteById(Long id) { securedFacilityRepository.deleteById(id); }
}

File: src/main/kotlin/com/example/eyeofsauron/service/SecuredFacilityService.kt
package com.example.eyeofsauron.service

import com.example.eyeofsauron.entity.SecuredFacility
import com.example.eyeofsauron.repository.SecuredFacilityRepository
import org.springframework.stereotype.Service

/**
 * Сервис для работы с охраняемыми объектами
 */
@Service
class SecuredFacilityService(private val repository: SecuredFacilityRepository) {
    fun getAll(): List<SecuredFacility> = repository.findAll()

    fun getById(id: Long) = repository.findById(id)

    fun create(securedFacility: SecuredFacility) = repository.save(securedFacility)

    fun update(securedFacility: SecuredFacility) {
        if(!repository.existsById(securedFacility.id))
            throw(Exception("Record not found!"))
        repository.save(securedFacility)
    }

    fun delete(securedFacility: SecuredFacility) = repository.delete(securedFacility)

    fun deleteById(id: Long) = repository.deleteById(id)
}